### DISCUSSION TITLE
Why was "shamefully-hoist" considered as a separate settings?

### DISCUSSION THREAD

**Discussion by is-it-ayush** [pnpm/pnpm#7498] [Q&A] [answered]
Hi! These might be stupid questions but I was wondering, 
- Why was `shamefully-hoist` given as a separate setting within `.npmrc` if settings it to `true` is the same as doing `public-hoist-pattern=*`? 
- Wouldn't setting `shamefully-hoist` to `false` be equivalent to `public-host-pattern=!*`?

From the [docs here](https://pnpm.io/npmrc#public-hoist-pattern),
> Note: Setting shamefully-hoist to true is the same as setting public-hoist-pattern to *.

**Comment by zkochan** [comment, marked as answer]
If I remember correctly, there was no `public-hoist-pattern` setting at first. We only had `shamefully-hoist`, which hoisted everything. Later we have added the possibility to selectively hoist dependencies.

Overall, I think it makes sense to have that setting as I assume most people won't bother to search for the right pattern, they just hoist everything.

**Reply by is-it-ayush** [reply]
Thank You! This makes sense to me. I believe giving the choice to selectively hoist packages was a "fix" to certain dependencies that weren't yet compatible with the virtual store architecture and still relied on a flat `node_modules` structure so as to retain the pnpm virtual store magic.
